protocol/src/main/java/org/fido/iot/protocol/To2ClientService.java

```diff
@@ -298,25 +299,48 @@ protected void doSetupDevice(Composite request, Composite reply) {
       newHash = getCryptoService().hash(hashType, secret, headerCopy.toBytes());
     }
 
-    Composite payload = Composite.newArray()
-        .set(Const.FIRST_KEY,
-            newHash != null ? newHash : PrimitivesUtil.getCborNullBytes());
+    Composite payload = Composite.newArray();
+    payload.set(
+        Const.FIRST_KEY,
+        newHash != null
+            ? newHash
+            : Composite.newArray().set(Const.FIRST_KEY, PrimitivesUtil.getCborNullBytes()));
+    payload.set(
+        Const.SECOND_KEY,
+        getStorage().getMaxOwnerServiceInfoMtuSz() != null
+            ? Integer.parseInt(getStorage().getMaxOwnerServiceInfoMtuSz())
+            : Composite.newArray().set(Const.FIRST_KEY, PrimitivesUtil.getCborNullBytes()));
```

Comment:
Serviceinfosize should be CBOR NULL, instead of an array of CBOR NULL.

Reply:
Updated.